enhancements/ingress/wildcard-routing.md

```diff
@@ -0,0 +1,189 @@
+no API exists to enable this feature on an ingress controller (i.e. router).
+
+OpenShift version 3 supports this feature through the `ROUTER_ALLOW_WILDCARD_ROUTES` environment variable when
+configuring the OpenShift Router directly. Users of this feature are blocked from upgrading to OpenShift 4 unless they
+discontinue use of the feature.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+See the [section][#wildcard-routing-policy]
+
+## Motivation
+
+See the [section][#wildcard-routing-policy]
+
+### Goals
+
+1. Define an API field that allows a cluster administrator to enable wildcard routing for an ingress controller.
+
+### Non-Goals
+
+1. Redesign how wildcard routing is implemented in [OpenShift Router](https://github.com/openshift/router).
+2. Introduce a separate API for managing wildcard routing.
+
+## Proposal
+
+Introduce a `WildcardRouting` field to the [IngressController](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)
+resource that allows administrators to enable or disable wildcard routing.
+
+### User Stories
+
+### Implementation Details/Notes/Constraints [optional]
+
+The [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) will use `WildcardRouting` to configure
+the environment variable `ROUTER_ALLOW_WILDCARD_ROUTES` in the [OpenShift Router](https://github.com/openshift/router).
+
+The default behavior should be to disallow wildcard routing, which is backwards compatible with existing
+`IngressController` resources.
+
+### Risks and Mitigations
+
+Addition of the new field requires no migration, as the default is not changing.
+
+## Design Details
+
+### New API Field
+
+`WildcardRouting` enum definition.
+
+```go
+// (Existing fields omitted, only newly proposed fields are visible)
+type RouteAdmissionPolicy struct {
+  // wildcardRouting enables or disables wildcard routing support for
+  // the ingress controller. WildcardRouting works in conjunction with
+  // the WildcardPolicy of routes [1] claimed by the ingress controller.
+  //
+  // [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+  //
+  // WildcardRouting can not be updated.
+  //
+  // If empty, defaults to "Disabled".
+  //
+  WildcardRouting *WildcardRoutingType `json:"wildcardRouting,omitempty"`
+}
+
+const (
+  // WildcardRoutingDisabled indicates only routes with a wildcard policy
+  // of "None" are admitted by the ingress controller.
+  WildcardRoutingDisabled WildcardRoutingType = "Disabled"
+
+  // WildcardRoutingEnabled indicates routes with any wildcard policy are
+  // admitted by the ingress controller.
+  WildcardRoutingEnabled WildcardRoutingType = "Enabled"
+)
```

Comment:
More policy-oriented values could be: WildcardsAllowed, WildcardsDisallowed

Reply:
Let's see what comes of https://github.com/openshift/enhancements/pull/252/files#r395760970. If the field name changes, I'll update the constants accordingly.

Reply:
done